### DISCUSSION TITLE
Modified F-16 pitch oscillation in Unreal Engine 5

### DISCUSSION THREAD

**Discussion by sajaxr** [JSBSim-Team/jsbsim#724] [Q&A]
Hello; there has already been discussion about oscillation in Unreal Engine (#695, #675, #629), but I could not find clear answers. 

There used to be pitch oscillation in the earlier version of the Unreal Engine plugin when reaching a certain speed. This was fixed with the update that set the plugin refresh rate to 120Hz, however, it returns if I try to modify the engine file F-16 uses to allow a higher maximum thrust value (and thus higher speed). Would someone know what could be causing this? I tried to set the project itself to a fixed framerate as well (30 and 60), but the oscillation still happened in a similar way even if those framerates were met constantly. The engine F-16 uses is called F100-PW-229, and I only changed the maximum thrust value. Already only knowing what can cause oscillation with high speeds could help. 

My ability to provide log files is highly limited, but I will do my best to give details if needed.

**Comment by seanmcleod** [comment]
My guess would be that the PID gains for the FCS aren't gain scheduled across the speed range and so you're potentially seeing oscillations as you get further and further away from where the gains have been tuned and tested at.
